lib/daemon.test.js

```diff
@@ -106,4 +107,61 @@ describe('Daemon', () => {
     expect(daemon._addPoller.called).to.equal(true)
     expect(daemon._removePoller.calledWith('test-id')).to.equal(true)
   })
+
+  it('manage externalsecrets with unmatched controller id and instance id', async () => {
+    const fakeExternalSecretEvents = (async function * () {
+      yield {
+        type: 'ADDED',
+        object: {
+          metadata: {
+            name: 'foo',
+            namespace: 'foo',
+            uid: 'test-id'
+          },
+          spec: {
+            controllerID: 'instance01'
+          }
+        }
+      }
+    }())
+
+    daemon._instanceId = 'instance01'
+    daemon._externalSecretEvents = fakeExternalSecretEvents
+    daemon._addPoller = sinon.mock()
+    daemon._removePoller = sinon.mock()
+
+    await daemon.start()
+    daemon.stop()
+
+    expect(daemon._addPoller.called).to.equal(true)
+    expect(daemon._removePoller.calledWith('test-id')).to.equal(true)
+  })
+
+  it('do not manage externalsecrets with unmatched controller id and instance id', async () => {
+    const fakeExternalSecretEvents = (async function * () {
+      yield {
+        object: {
+          type: 'ADDED',
```

Comment:
Looks like these lines are mixed up causing the test to pass.
```suggestion
        type: 'ADDED',
        object: { 
```